src/libopensc/card-dtrust.c

```diff
@@ -521,13 +999,34 @@ dtrust_decipher(struct sc_card *card, const u8 *data,
 static int
 dtrust_logout(sc_card_t *card)
 {
-	sc_path_t path;
+	struct dtrust_drv_data_t *drv_data;
 	int r;
 
-	sc_format_path("3F00", &path);
-	r = sc_select_file(card, &path, NULL);
+	SC_FUNC_CALLED(card->ctx, SC_LOG_DEBUG_VERBOSE);
 
-	return r;
+	drv_data = card->drv_data;
+
+	sc_sm_stop(card);
+	drv_data->pace = 0;
+	drv_data->can = 0;
```

Comment:
I would prefer using NULL here

Reply:
These are bit flags and not pointers. Are you sure to favor `NULL` here?